src/coreclr/jit/ee_il_dll.cpp

```diff
@@ -1615,45 +1615,40 @@ const char16_t* Compiler::eeGetShortClassName(CORINFO_CLASS_HANDLE clsHnd)
     return param.classNameWidePtr;
 }
 
-const WCHAR* Compiler::eeGetCPString(size_t strHandle)
+void Compiler::eePrintFrozenObjectDescription(const char* prefix, size_t handle)
 {
-#ifdef HOST_UNIX
-    return nullptr;
-#else
-    char buff[512 + sizeof(CORINFO_String)];
-
-    // make this bulletproof, so it works even if we are wrong.
-    if (ReadProcessMemory(GetCurrentProcess(), (void*)strHandle, buff, 4, nullptr) == 0)
-    {
-        return (nullptr);
-    }
-
-    CORINFO_String* asString = nullptr;
-    if (impGetStringClass() == *((CORINFO_CLASS_HANDLE*)strHandle))
+    const int maxStrSize = 64;
+    char16_t  str[maxStrSize];
+    int       realLength = this->info.compCompHnd->objectToString((void*)handle, str, maxStrSize);
+    if (realLength >= maxStrSize)
     {
-        // strHandle is a frozen string
-        // We assume strHandle is never an "interior" pointer in a frozen string
-        // (jit is not expected to perform such foldings)
-        asString = (CORINFO_String*)strHandle;
+        str[maxStrSize - 4] = L'.';
+        str[maxStrSize - 3] = L'.';
+        str[maxStrSize - 2] = L'.';
+        str[maxStrSize - 1] = 0;
     }
     else
     {
-        // strHandle is a pinned handle to a string object
-        asString = *((CORINFO_String**)strHandle);
+        str[realLength] = 0;
     }
 
-    if (ReadProcessMemory(GetCurrentProcess(), asString, buff, sizeof(buff), nullptr) == 0)
+    for (size_t i = 0; i < min(maxStrSize, realLength); i++)
     {
-        return (nullptr);
+        // Escape \n and \r symbols
+        if (str[i] == L'\n' || str[i] == L'\r')
+        {
+            str[i] = L' ';
+        }
     }
 
-    if (asString->stringLen >= 255 || asString->chars[asString->stringLen] != 0)
+    if (realLength >= 0)
     {
-        return nullptr;
+        printf("%s '%S'\n", prefix, str);
+    }
+    else
+    {
+        printf("%s 'frozen object handle'\n", prefix, str);
```

Comment:
```suggestion
        printf("%s 'frozen object handle'\n", prefix);
```
The formatting string is only using a single arg.